pkg/sql/schemachanger/scplan/internal/rules/current/dep_drop_object.go

```diff
@@ -34,41 +34,54 @@ import (
 func init() {
 
 	registerDepRule(
-		"descriptor TXN_DROPPED before DROPPED",
+		"descriptor dropped in transaction before removal",
 		scgraph.PreviousStagePrecedence,
-		"txn_dropped", "dropped",
+		"dropped", "absent",
 		func(from, to NodeVars) rel.Clauses {
 			return rel.Clauses{
 				from.TypeFilter(IsDescriptor),
 				from.El.AttrEqVar(screl.DescID, "_"),
 				from.El.AttrEqVar(rel.Self, to.El),
-				StatusesToAbsent(from, scpb.Status_TXN_DROPPED, to, scpb.Status_DROPPED),
+				StatusesToAbsent(from, scpb.Status_DROPPED, to, scpb.Status_ABSENT),
 			}
 		})
 
 	registerDepRule(
-		"descriptor DROPPED in transaction before removal",
-		scgraph.PreviousTransactionPrecedence,
-		"dropped", "absent",
+		"descriptor drop right before dependent element removal",
+		scgraph.Precedence,
+		"descriptor", "dependent",
 		func(from, to NodeVars) rel.Clauses {
 			return rel.Clauses{
 				from.TypeFilter(IsDescriptor),
-				from.El.AttrEqVar(screl.DescID, "_"),
-				from.El.AttrEqVar(rel.Self, to.El),
+				to.TypeFilter(IsSimpleDependent),
+				JoinOnDescID(from, to, "desc-id"),
 				StatusesToAbsent(from, scpb.Status_DROPPED, to, scpb.Status_ABSENT),
 			}
 		})
 
 	registerDepRule(
-		"descriptor drop right before dependent element removal",
+		"relation dropped before column no longer public",
 		scgraph.Precedence,
-		"descriptor", "dependent",
+		"descriptor", "column",
 		func(from, to NodeVars) rel.Clauses {
 			return rel.Clauses{
-				from.TypeFilter(IsDescriptor),
-				to.TypeFilter(IsSimpleDependent),
+				from.Type((*scpb.Table)(nil), (*scpb.View)(nil), (*scpb.Sequence)(nil)),
+				to.TypeFilter(IsColumn),
 				JoinOnDescID(from, to, "desc-id"),
-				StatusesToAbsent(from, scpb.Status_DROPPED, to, scpb.Status_ABSENT),
+				StatusesToAbsent(from, scpb.Status_DROPPED, to, scpb.Status_WRITE_ONLY),
+			}
+		})
+
+	registerDepRule(
```

Comment:
this one is interesting. Can you add commentary about why it matters? What's the hazard?

Reply:
The hazard is that the column in a dropped view may go to WRITE_ONLY before the view reaches DROPPED which is possible in an ALTER TABLE ... DROP COLUMN ... CASCADE which cascades to the view. This just never happened before but this rule and the one for indexes were always meant to be there. We might need one for constraints too when doing DROP CONSTRAINT.